server/api/users/users.controller.ts

```diff
@@ -29,9 +29,32 @@ export const getUserById: RequestHandler<{id: string}, UserProfile> = async (req
 };
 
 export const getUsers: RequestHandler<UserProfile[]> = async (req, res) => {
-  const users = await User.findAll({ attributes: User.allowedFields });
+  const page = req.query.page ? parseInt(req.query.page as string, 10) : 1;
```

Comment:
This may be open for debate, but I'd argue it should fallback to 1 if an invalid value is provided. If that's the case, we could rewrite this as:
```suggestion
  const page = parseInt(req.query.page as string, 10) || 1;
```

Reply:
i'll happily take the other side - if the user provides an invalid input, we should tell them so, otherwise we could be doing something wrong on the frontend but not realizing it because we are getting a 200 response from the server with actual data.

Reply:
I feel like most APIs I've encountered in the wild will just respond with the first page...

| Site | `page=1` or `start=0` | `page=2` or `start=1` | `page=foobar` or `start=foobar` |
| :--- | :--- | :--- | :--- |
| **Reques** | [Link](https://reqres.in/api/users?page=1) | [Link](https://reqres.in/api/users?page=2) | [Link](https://reqres.in/api/users?page=foobar) |
| **Unsplash** | [Link](https://unsplash.com/napi/search/photos?query=nerd&page=1) | [Link](https://unsplash.com/napi/search/photos?query=nerd&page=2) | [Link](https://unsplash.com/napi/search/photos?query=nerd&page=foobar) |
| **Newegg** | [Link](https://www.newegg.com/p/pl?d=floppy+disks&page=1) | [Link](https://www.newegg.com/p/pl?d=floppy+disks&page=2) | [Link](https://www.newegg.com/p/pl?d=floppy+disks&page=foobar) |
| **Unreal Engine Marketplace** | [Link](https://www.unrealengine.com/marketplace/api/assets?start=0) | [Link](https://www.unrealengine.com/marketplace/api/assets?start=1) | [Link](https://www.unrealengine.com/marketplace/api/assets?start=foobar) |

I tried to find random sites in different markets... notice how their invalid values still return the same results as their `page=0`/`start=1` while `page=1`/`start=2` returns the next set of results.

Reply:
I would argue that all those APIs are designed poorly.  APIs should not fail silently.

An example of a bad contract:
Client: give me the results for page 0
Server: here are the results for page 1
Client: give me the results for page 100
Server here are the results for page 1
Client: give me the results for page 1
Server: here are the results for page 1

An example of a better contract:
Client: give me the results for page 0
Server: we don't have a page zero, pages start at 1
Client: give me the results for page 100
Server: we don't have a page 100
Client: give me the results for page 1
Server: here are the results for page 1

Reply:
I mean, I don't disagree... but that doesn't appear to be convention unless I'm just looking at all the wrong sites.

[Even GitHub's API will default to page 1 when given an invalid page parameter](https://api.github.com/search/repositories?q=javascript&page=foobar), unless it's numerical and over their limit, then it returns an error. 🤦🏼‍♂️

The first site I found that doesn't do that is Amazon, but it doesn't error either... instead, it removes the `page` query param from the URL, thus [redirecting back to page 1](https://www.amazon.com/s?k=nerds&page=foobar). So, basically the same thing, just accomplished differently. 😂

Reply:
I'm going to agree with Tim on this. I think that a user requesting an invalid page should return a distinct error, not redirect to homepage.